packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -443,6 +518,54 @@ export default function(
     }
   }
 
+  function callGetDerivedStateFromProps(workInProgress, instance, props) {
+    const {type} = workInProgress;
+
+    if (typeof type.getDerivedStateFromProps === 'function') {
+      if (__DEV__) {
+        if (
+          typeof instance.componentWillReceiveProps === 'function' ||
+          typeof instance.UNSAFE_componentWillReceiveProps === 'function'
+        ) {
+          const componentName = getComponentName(workInProgress) || 'Unknown';
+          if (!didWarnAboutWillReceivePropsAndDerivedState[componentName]) {
+            warning(
+              false,
+              '%s: Defines both componentWillReceiveProps() and static ' +
+                'getDerivedStateFromProps() methods. We recommend using ' +
+                'only getDerivedStateFromProps().',
+              componentName,
+            );
+            didWarnAboutWillReceivePropsAndDerivedState[componentName] = true;
+          }
+        }
+      }
+
+      const partialState = type.getDerivedStateFromProps.call(
+        null,
+        props,
+        instance.state,
```

Comment:
We try to avoid reading things from the instance directly. Let's use `workInProgress.memoizedState` instead (probably should be passed as an argument to this function, same as props).

Reply:
Hm. We can't just use `workInProgress.memoizedState` here because it won't exist during the initialization path. So I can make `state` a param to this helper function, but I'll still have to pass `instance.state` down from `constructClassInstance`. That cool?